src/ServiceFabric/ServiceFabric/Commands/UpdateAzServiceFabricApplication.cs

```diff
@@ -0,0 +1,356 @@
+    {
+        [Parameter(Mandatory = true, Position = 0, ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Specify the name of the resource group.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty()]
+        public override string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, Position = 1, ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Specify the name of the cluster.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty()]
+        public override string ClusterName { get; set; }
+
+        [Parameter(Mandatory = true, Position = 2, ValueFromPipeline = true,
+                   HelpMessage = "Specify the name of the application")]
+        [ValidateNotNullOrEmpty()]
+        [Alias("ApplicationName")]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = false, Position = 3, ValueFromPipeline = true,
+                   HelpMessage = "Specify the application type version")]
+        [ValidateNotNullOrEmpty()]
+        public string ApplicationTypeVersion { get; set; }
+
+        [Parameter(Mandatory = false, Position = 4, ValueFromPipeline = true,
+                   HelpMessage = "Specify the application parameters as key/value pairs. These parameters must exist in the application manifest.")]
+        [ValidateNotNullOrEmpty()]
+        public Hashtable ApplicationParameter { get; set; }
+
+        [Parameter(Mandatory = false, ValueFromPipeline = true,
+                   HelpMessage = "Specifies the minimum number of nodes where Service Fabric will reserve capacity for this application")]
+        public long? MinimumNodes { get; set; }
+
+        [Parameter(Mandatory = false, ValueFromPipeline = true,
+                   HelpMessage = "Specifies the maximum number of nodes on which to place an application")]
+        public long? MaximumNodes { get; set; }
+
+        #region upgrade policy params
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter ForceRestart { get; set;}
+
+        [Parameter(Mandatory = false)]
+        public int? UpgradeReplicaSetCheckTimeoutSec { get; set;}
+
+        #region RollingUpgradeMonitoringPolicy
+
+        [Parameter(Mandatory = false)]
+        public FailureAction? FailureAction { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public int? HealthCheckRetryTimeoutSec { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public int? HealthCheckWaitDurationSec { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public int? HealthCheckStableDurationSec { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public int? UpgradeDomainTimeoutSec { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public int? UpgradeTimeoutSec { get; set;}
+
+        #endregion
+
+        #region ApplicationHealthPolicy
+
+        [Parameter(Mandatory = false)]
+        public bool? ConsiderWarningAsError { get; set; }
```

Comment:
it should be a switch parameter, see https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/design-guidelines/parameter-best-practices.md#bool-vs-switchparameter